Update to setup documentation

meg-gutshall opened the pull request:
# Description
I went through the setup process in the `CONTRIBUTING.md` file and noticed some errors.

The commands were running out of the `coral` directory instead of the `coral/rails` directory which was giving back errors.

Also, I could not get the database to initialize because it couldn't find the POSTGRES_DB key. I was looking for a `.env` template and couldn't find one so decided to make one myself and make a few changes to the guide. If any other variables need to be added for the Docker setup (i.e. Sidekiq), please stick them in.

# Type of change
Documentation update

# Screenshot
![Setup Issues Screenshot](https://user-images.githubusercontent.com/37842352/66350054-465d6400-e928-11e9-8f56-b343fd521e4c.png)

meg-gutshall committed:
Create .env template

Created a .env template for those who are new to using Postgres or having trouble starting the server. My Postgres database would not create itself until I added the POSTGRES_DB key into the .env file. I could see a lot of users not figuring that out.

meg-gutshall committed:
Remove files from .gitignores

Removed the .env* filenames from both .gitignore files to allow for the addition of the .env.example file. Kept the .env filename in both .gitignore files. Also removed some duplicate filenames from the rails/.gitignore.

meg-gutshall committed:
Adjust contributing guide

Made contributing guide more beginner-friendly by being more specific in how to run tests. Also directions were a little off in that users are told to run commands in the coral directory instead of the coral/rails directory. I also added a step that includes creating a .env file and copying the body of the newly added .env.example file to set up the user's Postgres database.

cflipse commented on .gitignore:
```diff
@@ -8,7 +8,7 @@
 
 # Ignore .dotenv config
-.env*
+.env
```
leave the * there please; `.env.development` etc is valid for environment specific `.env` files and the wildcard will catch all of those

meg-gutshall replied:
Okay! I'm submitting a revision right now.

meg-gutshall committed:
Implement requested PR review changes

meg-gutshall committed:
Implement requested PR review changes

I tried to separate out and label some of the patterns to make it more readable. I also noticed that some of the duplicate patterns I had previously deleted weren't exact duplicates, so I consulted the Git docs and left the duplicate pattern with the broader scope (i.e. `node_modules` instead of `/node_modules`).

meg-gutshall commented:
I changed both `.gitignore` files to reflect the following patterns:
```
$ rails/.gitignore
.env*
!.env.example

$ .gitignore
.env*
!rails/.env.example
```
This should ignore all `.env` files except the example file which I created.

I sectioned off a few of the other patterns to make them more readable. In addition, I noticed that some of the duplicate patterns I had previously deleted weren't **_exact_** duplicates, so I consulted the Git docs and left the duplicate pattern with the broader scope (i.e. `node_modules` instead of `/node_modules`).
